### PR TITLE
feat: Add feedback button

### LINKED ISSUE #105: Share log messages
Dependent on #103 and #83.

We need a way to share log files from the TenTenOne App with us. using FLog we can make use of the `FLog.exportLogs()`

### LINKED ISSUE #414: Add send feedback button from within the app
The user should be able to send feedback to the app.

**Acceptance Criterias**

- [x] The button click should open an email template on the users phone
- [x] The logs should be automatically exported to file (zipped?) and attached to the email

### PR DESCRIPTION
**Pull request by holzeis**
I found this pretty cool feedback package (https://pub.dev/packages/feedback), that will allow the user to draw on the screen, add a feedback and send his logs via email.

I guess we we will not use that on a major release, but to get feedback on a beta version, it's pretty useful.

Works on Android and ios (not couldn't test on ios, since the simulator does not come with an email client). Updating the text "What's wrong" is kind of possible, but really ugly through translations. I tried it, but stopped again as it wasn't worth it. If you are very much bothered about it feel free to create a follow up PR 🙈 .

I struggled a bit with where to put that button. I could think of the following.

1. always visible: but where, it shouldn't distract too much.
2. in the drawer: i guess pretty hidden and the user might not find it.
3. on the wallet dashboard: visible at the beginning but moves out of sight the more the app is used. The user should still know that it's there though.

resolves #105 
resolves #414 

feedback

### CODE REVIEW

File: lib/wallet/wallet_dashboard.dart [hunk 1]
```diff
@@ -92,6 +98,31 @@ class _WalletDashboardState extends State<WalletDashboard> {
...
+        child: const Text('Provide feedback'),
+        onPressed: () {
+          BetterFeedback.of(context).show((feedback) async {
+            final screenshotFilePath = await writeImageToStorage(feedback.screenshot);
+            final logs = await FLog.exportLogs();
+
+            final Email email = Email(
+              body: feedback.text,
+              subject: '10101 Feedback',
+              recipients: ['richard@coblox.tech'],
```
[the remark this answers is not in the document]
Comment (bonomat) [reply lib/wallet/wallet_dashboard.dart:135]: I've created `10101-devs@coblox.tech` for now. We don't have emails for 10101.finance yet.

Comment (bonomat) [inline lib/wallet/wallet_dashboard.dart:135]:
```suggestion
              recipients: ['10101-devs@coblox.tech'],
```

**Review by bonomat — APPROVED** [review]
Awesome find. I would also leave this plugin for the major release as we want to be able to collect feedback for quite a while :)

Re positioning, I would have put it into the menu/drawer.

**Review by luckysori — APPROVED** [review]
Awesome find! I would also put it in the drawer.

File: lib/wallet/wallet_dashboard.dart [hunk 2]
```diff
@@ -92,6 +99,49 @@ class _WalletDashboardState extends State<WalletDashboard> {
     );
 
     widgets.add(paymentHistoryList);
+    if (Platform.isAndroid || Platform.isIOS || true) {
```
Comment (klochowicz) [inline lib/wallet/wallet_dashboard.dart:102]:
```suggestion
```
no need for this `if` anymore, if it always evaluates to `true` 😉

**Review by klochowicz — APPROVED** [review]
great find!

**Comment by klochowicz** [comment]
I will test it on iOS right now.

**Comment by da-kami** [comment]
> Awesome find! I would also put it in the drawer.

Given that the number one purpose for this release is to collect feedback I'd definitely not hide the feedback button in the drawer. I think It's good on the dashboard because it is yet another action that we would expect the user to do. The look and feel being different than the action cards is good too from my perspective. I think it is good the way it is :)

**Review by da-kami — APPROVED** [review]
Awesome Flutter plugin! 
Makes me consider to become a sponsor :D

**Comment by da-kami** [comment]
> > > Awesome find! I would also put it in the drawer.
> > 
> > 
> > Given that the number one purpose for this release is to collect feedback I'd definitely not hide the feedback button in the drawer. I think It's good on the dashboard because it is yet another action that we would expect the user to do. The look and feel being different than the action cards is good too from my perspective. I think it is good the way it is :)
> 
> I think we could do both. you would like to give feedback from other places, and dashboard is pretty haphazard.

True! I'm in favor for adding it to the drawer menu in addition to keeping it on the Dashboard :)

File: lib/wallet/wallet_dashboard.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (holzeis) [reply lib/wallet/wallet_dashboard.dart:135]: thanks

File: lib/wallet/wallet_dashboard.dart (hunk 2 above)
Comment (holzeis) [reply lib/wallet/wallet_dashboard.dart:102]: ups.. I added true only for testing 😅 good catch 👍

**Comment by holzeis** [comment]
I added it to the drawer. It makes a lot of sense there since the drawer is always available and by using `Navigator.pop()` you will be able to give feedback directly at the page you are currently at.

**Review by luckysori — APPROVED** [review]

**Pull request merged**